### PR TITLE
fix: Use HttpRequest authority instead of host

### LINKED ISSUE #6878: Use `authority()` instead of deprecated `host()` method in "allow list"
### Description
Besu code uses `RoutingContext.request.host()` to determine the host header which is then used to determine the "host allow list" logic. This method has been deprecated in Vert.x. It should be replaced with `RoutingContext.request.authority()`. This would allow a more cleaner logic and future proofing the code. 

### Acceptance Criteria
Fix at following locations:
org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpService.java:224
org/hyperledger/besu/ethereum/api/jsonrpc/EngineJsonRpcService.java:622
org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpService.java:513
org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketService.java:297
org/hyperledger/besu/metrics/prometheus/MetricsHttpService.java:161

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR description
fix: Use HttpRequest authority instead of host.
- Use authority method to determine the host name from header instead of using deprecated host method
- Fix WebSocketService unit test

## Fixed Issue(s)

Fixes #6878 

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [x] acceptance tests: `./gradlew acceptanceTest`
- [x] integration tests: `./gradlew integrationTest`
- [x] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketHostAllowlistTest.java [hunk 1]
```diff
@@ -200,10 +217,18 @@ private void doHttpRequestAndVerify(
...
+                      assertThat(response.result().statusCode()).isEqualTo(expectedResponse);
+                      testContext.completeNow();
+                    } else {
+                      testContext.failNow(response.cause());
+                    }
                   });
         });
-    testContext.awaitCompletion(VERTX_AWAIT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    assertThat(testContext.awaitCompletion(VERTX_AWAIT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS))
+        .isTrue();
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketHostAllowlistTest.java:229]: @usmansaleem is this what you mean by the test needing fixing? there was no assertion before?

**Review by macfarla — COMMENTED** [review]
it is much simpler!

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketHostAllowlistTest.java (hunk 1 above)
Comment (usmansaleem) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/websocket/WebSocketHostAllowlistTest.java:229]: @macfarla correct,  from L220 and L232 as well.

**Review by jflo — APPROVED** [review]
better exception handling and much less copypasta!

**Pull request merged**